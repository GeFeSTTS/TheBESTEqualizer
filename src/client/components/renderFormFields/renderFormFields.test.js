import checkPropTypes from 'check-prop-types';
import React from 'react';
import { shallow } from 'enzyme';
import RenderFormFields from './renderFormFields';

const defaultProps = {
  onInputChange: jest.fn(),
  fieldsToRender: [{
    name: '',
    label: '',
    type: '',
  }],
  userData: {},
  validationErrors: {},
};

const checkProps = (component, expectedProps) => {
  const propError = checkPropTypes(component.propTypes, expectedProps, 'prop', component.name);
  expect(propError).toBeUndefined();
};

const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<RenderFormFields {...setupProps} />);
};

describe('rendering component', () => {
  it('render login form field without error', () => {
    const fieldsToRender = [{
      name: 'email',
      label: 'Email',
      type: 'text',
    },
    {
      name: 'password',
      label: 'Password',
      type: 'text',
    }];
    const wrapper = setup({ fieldsToRender });
    expect(wrapper.find('FormField')).toHaveLength(2);
  });
  it('render registration form field without error', () => {
    const fieldsToRender = [{
      type: 'text',
      name: 'username',
      label: 'Username',
    },
    {
      name: 'email',
      label: 'Email',
      type: 'text',
    },
    {
      name: 'password',
      label: 'Password',
      type: 'password',
    },
    {
      type: 'password',
      name: 'passwordConfirmation',
      label: 'Pasword confirmation',
    }];
    const wrapper = setup({ fieldsToRender });
    expect(wrapper.find('FormField')).toHaveLength(4);
  });
});

describe('testing ptopTypes', () => {
  it('does not throw warning with expected props', () => {
    checkProps(RenderFormFields, defaultProps);
  });
});

describe('snapshot', () => {
  it('should render formField properly', () => {
    const wrapper = setup();
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
