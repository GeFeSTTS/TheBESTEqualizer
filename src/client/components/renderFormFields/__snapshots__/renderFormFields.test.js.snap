// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should render formField properly 1`] = `"<FormField onInputChange={[Function: mockConstructor]} field={{...}} value=\\"\\" error=\\"\\" />"`;
